project-docs/javascript-linter-spec.md

```diff
@@ -0,0 +1,178 @@
+
+### Optional ingredients
+
+A recipe can mark an ingredient as optional by placing a question mark after it, like `prose.description?`. This means that the page does not have to contain a representation of the ingredient, but if it does, the representation must be properly formed according to the ingredient's specification, and the representation must appear in the place specified by the recipe.
+
+Technically, whether an ingredient is optional or not is a separate attribute of the ingredient. This means that a single ingredient may be marked as optional in one recipe, and mandatory in a different recipe. In practice, in the JavaScript documentation, this does not occur: there are currently no ingredients that are optional in some recipes and mandatory in others.
+
+### Page sections
+
+Many, but not all, ingredients correspond to sections of a page. A section of a page is defined as a part of the page beginning with an `<h2>` heading and ending with either another `<h2>` or the end of the page.
+
+Sections are identified by an `id` attribute on the `<h2>`. For example, the `prose.description` section starts with an element whose opening tag is like `<h2 id="Description">`. In this document we'll refer to page sections using a notation like `h2#identifier`: for example, `h2#Description`.
+
+## JavaScript recipes
+
+We've defined the following recipes for the JS docs:
+
+- [javascript-class](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-class.yaml)
+- [javascript-namespace](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-namespace.yaml)
+- [javascript-constructor](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-constructor.yaml)
+- [javascript-method](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-method.yaml)
+- [javascript-property](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-property.yaml)
+- [javascript-error](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-error.yaml)
+- [javascript-language-feature](https://github.com/mdn/stumptown-content/blob/master/recipes/javascript-language-feature.yaml)
+
+## JavaScript recipe ingredients
+
+In total these recipes use the following ingredients:
+
+prose ingredients:
+
+- prose.short_description
+- prose.description?
+- prose.see_also
+- prose.syntax
+- prose.message
+- prose.error_type
+- prose.what_went_wrong
+- prose.\*
+
+data ingredients:
+
+- data.interactive_example?
+- data.class_constructor
+- data.constructor_properties?
+- data.static_properties?
+- data.static_methods?
+- data.instance_properties?
+- data.instance_methods?
+- data.examples
+- data.specifications
+- data.browser_compatibility
+
+All ingredients except two are demarcated in the page using an `<h2>` heading. The exceptions are `prose.short_description` and `data.interactive_example`.
```

Comment:
Thanks for the comment Joe. I somehow missed your bug report although you did tell me about it. I've commented there.

I think you know this too, but to be clear: the linter you're referring to in https://github.com/mdn/stumptown-content/issues/375 is the tool that lints Markdown in GitHub. This linter is not being actively developed right now, because we don't at the moment have a roadmap to get content into Markdown in GitHub.

This is distinct from the linter being discussed in this thread, which lints MDN Wiki content, and which *is* under active development.